wordpress-mobile/WordPress-iOS
# [Modal Layout Picker] "Preview" button functionality

**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2452

**Related-PRs**
- `gutenberg` https://github.com/WordPress/gutenberg/pull/25200
- `gutenberg-mobile` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2606

## To test:
🗒️ These are the same testing steps in the related PRs
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

- Start by navigating to the Modal Layout Picker

### Preview a layout
1. Select a layout from the list
1. Select Preview
- **Expect** to see a read-only editor load that shows a preview of the selected layout.

### Preview a layout, then a different one
1. Select a layout from the list
1. Select Preview
- **Expect** to see a read-only editor load that shows a preview of the selected layout.
1. Select Back
1. Select a different layout from the list
1. Select Preview
- **Expect** to see a read-only editor load that shows a preview of the newly selected layout.

### Preview a layout, then cancel
1. Select a layout from the list
1. Select Preview
- **Expect** to see a read-only editor load that shows a preview of the selected layout.
1. Select Back
- Or close the modal by swiping down from the top
1. Close the modal via the ✖️ 

### Preview a layout, then create a page
1. Select a layout from the list
1. Select Preview
- **Expect** to see a read-only editor load that shows a preview of the selected layout.
1. Select Create Page
- **Expect** to see the page editor load with the selected layout content populated.
- **Expect** to see the page editor load with a title that relates to the content. Example a layout from the "About Pages" should have a title like "About"

### Spot check for regressions
Open an existing page or blog, expect the content to be loaded and for normal editing to take place.

### Spot check Modal Layout Picker Disabled
1. Disable Modal Layout Picker
1. Create a new page
- **Expect** to see the existing Starter Page Template Picker
1. Select an option
- **Expect** to see the Preview with "Close" and "Apply" as actions
1. Try Each option, expect it to behave as it previously did. 

## Screenshots 
![Preview 2020-09-09 16_08_26](https://user-images.githubusercontent.com/3384451/92647969-c0c00200-f2b6-11ea-9eeb-84aa515f1d93.gif)

## PR submission checklist:
